Update dac documentation

TucksonDev opened the pull request:
This PR updates the documentation available to configure a DAC.

Important notes:
- Multiple partial views were created for this work. There's currently a [limitation with Docusaurus](https://github.com/facebook/docusaurus/issues/3915) that doesn't show headings of imported MD documents in the TOC of the right sidebar. Because the two main guides share a fair amount of content, and to avoid duplication as much as possible, partial views have been created for the content that is shared among both guides. This partial views don't include the headers of the sections, which are added directly to the main document to have them available in the TOC. Once that limitation is solved, we can reduce the number of partial views.
- One of the styles has been slightly adjusted to allow for `tabgroups` without a "label".
- Last page is not present as it is being worked on in this PR: #799 . An appropriate banner has been placed instead.
- Placement of the section can be changed. It is now placed at the end of the Node running section, but can be moved somewhere else. We can prioritize publishing the doc before finding the best place for this docs too.

[Preview](https://nitro-docs-git-update-dac-documentation-offchain-labs.vercel.app/node-running/how-tos/data-availability-committee/introduction)

symbolpunk commented:
In review

Tristan-Wilson commented on arbitrum-docs/node-running/how-tos/data-availability-committee/introduction.mdx:
```diff
@@ -0,0 +1,56 @@
+  - Public BLS key
+  - The https URL for the RPC endpoint which includes some random string (e.g. das.your-chain.io/rpc/randomstring123), communicated through a secure channel
+  - The https URL for the REST endpoint (e.g. das.your-chain.io/rest)
+- [Deploy a mirror DAS](./deploy-a-mirror-das.mdx) if your main DAS doesn't have the REST interface enabled. Send the following information to the chain owner:
```
The reason for using a mirror DAS is more along the lines of if you don't want to expose the REST interface of your main DAS to the public. The mirror DAS still needs the REST interface on the DAS to sync from it. The reason deployers might want to do this would be to separate out the server that is handling the Sequencer API from the public REST interface for resiliency, eg in case of a DDoS attack on the REST interface.

TucksonDev replied:
Alright, I've modified the language to make it clearer than the main use case for running a mirror DAS is to complement your main DAS.

Tristan-Wilson commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_1-how-das-work-content.mdx:
```diff
@@ -0,0 +1,10 @@
+
+DA servers accept time-limited requests to store data batches from the sequencer of an AnyTrust chain, and return a signed certificate promising to store that data during the established time. They also respond to requests to retrieve the data batches.
+
+Additionally, mirror DA servers can be run to help handle requests to retrieve data batches (however, they won't store batches). They may also provide archived data beyond the limited time that DA servers are required to store the data.
```
Maybe '(however, they won't handle requests to store batches)', since they do store the batches but don't support the Sequencer API.

Tristan-Wilson commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.1-das-configuration-options-interfaces.mdx:
```diff
@@ -0,0 +1,13 @@
+There are two main interfaces that can be enabled in a DAS: an **RPC interface** to store data in the DAS, intended to be used only by the AnyTrust sequencer; and a **REST interface** that supports only GET operations and is intended for public use.
```
I don't know if we're bothering with the Sequencer/Batch Poster distinction here. They can be configured to run on the same instance of Nitro or separate instances.

TucksonDev replied:
I've only used Sequencer for DAS deployers and Batch poster for DAC chain owners. They are in different guides aimed at different audiences, but thought that those would be the terms that each audience are most familiar with.

Tristan-Wilson commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.1-das-configuration-options-interfaces.mdx:
```diff
@@ -0,0 +1,13 @@
+
+:::info
+
+Notice that it is possible to run one DAS that enables both interfaces, or one DAS with only the RPC interface enabled with an additional mirror DAS to serve REST requests.
```
Same comment as before about the DAS needing to run the REST interface for the mirror to sync from, but maybe it's not exposed publicly. (Unless using the same S3 bucket backend for DAS and mirror, or using IPFS.)

TucksonDev replied:
Modified the language to try to express this.

Tristan-Wilson commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.2-das-configuration-options-storage.mdx:
```diff
@@ -0,0 +1,10 @@
+
+If more than one option is selected, store requests must succeed to all of them for it to be considered successful, while retrieve requests only require one of them to succeed.
+
+If there are other storage backends you'd like us to support you can send us a message on [Discord](https://discord.gg/arbitrum), or use the “Request an update” button on top of this page.
```
It's also (in theory) fairly easy for people to add support for backends they are interested in and contribute it back to the Nitro repo. We have one third party contribution in the works currently for this, see the below issue and PR:

* https://github.com/OffchainLabs/nitro/issues/1989#issuecomment-1832540609
* https://github.com/OffchainLabs/nitro/pull/1992

TucksonDev replied:
Thanks 💪 . I've added that as part of the paragraph, instead of just requesting an update. [Here](https://nitro-docs-git-update-dac-documentation-offchain-labs.vercel.app/node-running/how-tos/data-availability-committee/deploy-a-das#storage-options).

TucksonDev commented:
Thanks @Tristan-Wilson !
I've added a few more changes to the guides:
- The main guide now doesn't mention the possibility of "disabling" the REST interface, but rather talks about [running a mirror DAS](https://nitro-docs-git-update-dac-documentation-offchain-labs.vercel.app/node-running/how-tos/data-availability-committee/deploy-a-das#running-a-mirror-das) to support your main DAS in handling REST requests.
- Simplified the [mirror DAS guide](https://nitro-docs-git-update-dac-documentation-offchain-labs.vercel.app/node-running/how-tos/data-availability-committee/deploy-a-mirror-das) since it's now aimed at members who already have a main DAS running. Added a notice on top and redirections of the introductory sections to the main guides.
- Rearranged some of the content to support those 2 changes (e.g., adding the "Archive" section to both guides.

Note: I've kept the partial views for now, until everything is good. Will remove those before merging.

Tristan-Wilson reviewed (approved):
LGTM

symbolpunk commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.3-das-configuration-options-caching.mdx:
```diff
@@ -0,0 +1,3 @@
+An in-memory cache can be enabled to avoid needing to access underlying storage for retrieve requests.
+
+Requests sent to the REST interface (to retrieve data from the DAS) return always the same data for a given hash so the result is cacheable. It also contains a `cache-control` header specifying that the object is immutable and to cache it for up to 28 days.
```
```suggestion
Requests sent to the REST interface (to retrieve data from the DAS) always return the same data for a given hash, so the result is cacheable. It also contains a `cache-control` header specifying that the object is immutable and to cache it for up to 28 days.
```

symbolpunk commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.4-das-configuration-options-state-sync.mdx:
```diff
@@ -0,0 +1 @@
+DA servers also have an optional REST aggregator which, when a data batch is not found in cache or storage, requests that batch to other REST servers defined in a list and stores that batch upon receiving it. This is how a DAS that misses storing a batch (the AnyTrust protocol doesn't require all of them to report success in order to post the batch's certificate to the parent chain) can automatic ...
```
Flag - this partial can be used above (seeing duplication of this paragraph in `deploy-a-das`)

symbolpunk reviewed (approved):
This is a beast of a PR... lots of misc nits, no blocking feedback. TYVM 🙏

TucksonDev commented on arbitrum-docs/node-running/how-tos/data-availability-committee/configure-the-dac-in-your-chain.mdx:
```diff
@@ -0,0 +1,12 @@
+title: 'How to configure a Data Availability Committee: configure the DAC in your chain'
+description: This how-to will help you configure the DAC in your chain.
+author: jose-franco
+sidebar_label: Configure the DAC in your chain
```
Modified the sidebar to "How to configure the Data Availability Committee (DAC) in your chain" since this document is specific about one step of the process and having the same title as the section might be confusing for users. I kept the suggested convention.

TucksonDev commented on arbitrum-docs/node-running/how-tos/data-availability-committee/configure-the-dac-in-your-chain.mdx:
```diff
@@ -0,0 +1,12 @@
+---
+title: 'How to configure a Data Availability Committee: configure the DAC in your chain'
```
Modified the title too, as specified [here](https://github.com/OffchainLabs/arbitrum-docs/pull/882#discussion_r1467457774).

TucksonDev commented on arbitrum-docs/node-running/how-tos/data-availability-committee/deploy-a-das.mdx:
```diff
@@ -0,0 +1,443 @@
+<PublicPreviewBannerPartial />
+
+<p>
+  <a data-quicklook-from="arbitrum-anytrust-protocol">AnyTrust</a> chains rely on an external Data
```
100%. I usually do the same too, but after so many changes I forget to do a final review of these conventions 🙈 
Thanks for the flag!

TucksonDev commented on arbitrum-docs/node-running/how-tos/data-availability-committee/deploy-a-mirror-das.mdx:
```diff
@@ -0,0 +1,285 @@
+
+In this how-to you'll learn how to deploy a mirror DAS and enable a REST interface to respond to requests of stored information. For more information related to configuring a DAC, please see the [Introduction](./introduction.mdx).
+
+You should be familiarized with how the AnyTrust protocol works and what's the role of the DAC in the protocol. You can find more information about the AnyTrust protocol in [Inside AnyTrust](/inside-anytrust.mdx). You should also be familiarized with how a regular DAS works and what configuration options are available. You can find that information in [How to deploy a DAS](./deploy-a-das.mdx). Fi ...
```
Following the convention established in the prereq document, I've changed it to:

This how-to assumes that you're familiar with:

- How a regular DAS works and what configuration options are available. Refer to [How to deploy a DAS](https://nitro-docs-git-update-dac-documentation-offchain-labs.vercel.app/node-running/how-tos/data-availability-committee/deploy-a-das) for a refresher.
- [Kubernetes](https://kubernetes.io/). The examples in this guide use Kubernetes to containerize your DAS.

TucksonDev commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.2-das-configuration-options-storage.mdx:
```diff
@@ -0,0 +1,10 @@
+- [Badger](https://dgraph.io/docs/badger/) database on local disk
+- [IPFS](https://ipfs.tech/)
+
+If more than one option is selected, store requests must succeed to all of them for it to be considered successful, while retrieve requests only require one of them to succeed.
```
These files were temporarily kept in case I needed to rescue them again (they were used to hold the common content between both DAS guides). I've now deleted them since they are not used anymore.
Thanks for the flag!

TucksonDev commented on arbitrum-docs/node-running/how-tos/data-availability-committee/partials/_2.1-das-configuration-options-interfaces.mdx:
```diff
@@ -0,0 +1,13 @@
+
+:::
+
+Finally, IPFS is an alternative interface that serves batch retrieval. A mirror DAS can be configured to sync and pin batches to its local IPFS repository, then act as a node in the IPFS peer-to-peer network. The advantage of using this interface is that a Nitro node that is configured to use IPFS will use the batch hashes to find the batch data on the IPFS peer-to-peer network. Depending on the  ...
```
[See comment here](https://github.com/OffchainLabs/arbitrum-docs/pull/882#discussion_r1467468284)
